File: backend/src/common/password-policy.ts
import { BadRequestException } from '@nestjs/common';

/**
 * Política de complejidad de contraseña (RNF_S_01 / CU-06 Flujo Alt. 3.2).
 * Exige al menos 8 caracteres y un número. Compartida por los flujos de
 * registro (CU-06), recuperación (CU-03) y actualización de perfil (CU-07).
 */
export function validatePasswordPolicy(password: string): void {
  const hasMinimumLength = password.length >= 8;
  const hasNumber = /\d/.test(password);
  if (!hasMinimumLength || !hasNumber) {
    throw new BadRequestException('La contraseña debe tener al menos 8 caracteres y un número');
  }
}

File: frontend/lib/features/auth/presentation/validators/password_policy.dart
/// Políticas de seguridad de credenciales (CU-06 / RNF_S_01).
///
/// La especificación exige contraseñas con al menos 8 caracteres y un número:
/// "La contraseña debe tener al menos 8 caracteres y un número".
class PasswordPolicy {
  static const int minLength = 8;
  static const String errorMessage =
      'La contraseña debe tener al menos 8 caracteres y un número';

  /// Retorna `true` si la contraseña cumple la política (>= 8 caracteres y al
  /// menos un dígito).
  static bool isValid(String password) {
    if (password.length < minLength) return false;
    return password.contains(RegExp(r'[0-9]'));
  }
}